src/renderer/vt/Xterm256Engine.cpp

```diff
@@ -128,6 +131,34 @@ Xterm256Engine::Xterm256Engine(_In_ wil::unique_hfile hPipe,
     return S_OK;
 }
 
+// Routine Description:
+// - Write a VT sequence to start/stop a hyperlink
+// Arguments:
+// - textAttributes - Text attributes to use for the hyperlink ID
+// - pData - The interface to console data structures required for rendering
+// Return Value:
+// - S_OK if we succeeded, else an appropriate HRESULT for failing to allocate or write.
+HRESULT Microsoft::Console::Render::Xterm256Engine::_UpdateHyperlinkAttr(const TextAttribute& textAttributes,
+                                                                         const gsl::not_null<IRenderData*> pData) noexcept
+{
+    if (textAttributes.IsHyperlink() != _lastTextAttributes.IsHyperlink())

```

Comment:
will this work when we have two back-to-back hyperlinks with different URI?

Maybe all we need to do is check if their hyperlink IDs match.. then we get automatic detection for going from 0->anything, anything->0, and anything->anything_else